335 bring `teal_slice` and `teal_slices` to package INDEX

m7pr opened the pull request:
Proposition on how to solve #335

- Removed `@rdname teal_slice` from utility functions for `teal_slice` 
- Created customized `@usage` section to display both `teal_slice` and utility functions on the same page
- Result: `teal_slice` appearing in package index, where utility functions only appear in `teal_slice` documentation page

If this is acceptable, I will do the same for `teal_slices`

<img width="946" alt="index" src="https://github.com/insightsengineering/teal.slice/assets/133694481/d1b3f113-a099-4486-a603-bd8cfdd0c44e">
<img width="935" alt="documentation" src="https://github.com/insightsengineering/teal.slice/assets/133694481/e07b5fce-61b5-4c82-bcd7-fde2bd2fe9f9">

chlebowa reviewed (commented):
Excellent! :ok_hand: Please proceed with `teal_slices`.

I would just add a comment line to explain why `@usage` is used manually, in case we forget in the future.

m7pr commented:
Hey @chlebowa proceeded with `teal_slices` and added comments above `@usage` section with the motivation for it.
Also added a comment in function's definitions to remember to update the `@usage` is parameters are edited.

m7pr commented:
Will also need to move `@param` documentation from `format.teal_slices` to `teal_slices` and fix R CMD CHECKS.

chlebowa commented:
Did you forget to push your changes?

m7pr commented:
I think I pushed them, but on wrong branch (334_teal_slice_index@main instead of 335_teal_slice_index@main). Will push here.

m7pr commented:
Should be visible on this branch. Sorry! 

PS. I will also delete `334_teal_slice_index@main`.

chlebowa reviewed (commented):
Please place the `@usage` section between `@details` and `@params` to reflect the structure of the man page better.

Any ideas how to tackle the check warnings? 

```
  Undocumented code objects:
    ‘as.teal_slice’ ‘as.teal_slices’ ‘is.teal_slice’ ‘is.teal_slices’


  Functions or methods with usage in documentation object 'teal_slices' but not in code:
    ‘as.list’ ‘[’ ‘c’ ‘format’ ‘print’


  Objects in \usage without \alias in documentation object 'teal_slice':
    ‘is.teal_slice’ ‘as.teal_slice’ ‘as.list.teal_slice’
    ‘format.teal_slice’ ‘print.teal_slice’
  
  Objects in \usage without \alias in documentation object 'teal_slices':
    ‘is.teal_slices’ ‘as.teal_slices’ ‘as.list’ ‘[’ ‘c’ ‘format’ ‘print’
```

m7pr commented:
> Please place the @usage section between @details and @params

I moved it between `@description` and `@params`. Description comes before the usage, then we have Usage and after that you see Arguments in the documentation page.

chlebowa commented:
I tied my hand at removing those warnings and I was only able to remove one. I hope you will find a way.

m7pr commented:
I added `@aliases` and updated `@usage` section for S3 methods based on this thread https://stackoverflow.com/a/11285863

m7pr commented:
OK, this was a bumpy ride, but I figured this out.

Seeing no warnings locally during R CMD CHECK

```r
   Status: OK
   
── R CMD check results ───────────────────────────────────────────────────────────────────────────────────────── teal.slice 0.5.0.9004 ────
Duration: 2m 11.8s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔

R CMD check succeeded
```

This should be enough to get merged. Pretty cool learning exercise about custom `@usage` section.

chlebowa commented on R/teal_slice.R:
```diff
@@ -2,6 +2,35 @@
+#'
+#' as.teal_slice(x)
+#'
+#' \\method{as.list}{teal_slice}(x, ...)
```
I assume this cannot be done with `@method as.list teal_slice` because `@usage` is built manually?

chlebowa reviewed (changes requested):
![image](https://github.com/insightsengineering/teal.slice/assets/114988527/69431fd2-8733-4346-a405-baa45bfe8b9d)

m7pr commented:
doh

chlebowa reviewed (approved):
Nice job :ok_hand:

chlebowa commented:
> But now I do see `as.teal_slice(s)` and `is.teal_slice(s)` in package index. So this needs a bit more investigation.

So are all the methods :frowning_face:

chlebowa reviewed (changes requested):
Not quite there yet.

m7pr commented:
So adding `@aliases` like

```r
#' @aliases is.teal_slices as.teal_slices as.list.teal_slices format.teal_slices print.teal_slices c.teal_slices [.teal_slices
```

list those functions as topics in package index. 

But if you do not have those, then R CMD CHECK throws:

```r
W  checking for missing documentation entries (1.3s)
   Undocumented code objects:
     'as.teal_slices' 'is.teal_slices'
   All user-level objects in a package should have documentation entries.
   See chapter 'Writing R documentation files' in the 'Writing R
   Extensions' manual.
✔  checking for code/documentation mismatches (3.1s)
W  checking Rd \usage sections (854ms)
   Objects in \usage without \alias in documentation object 'teal_slices':
     'is.teal_slices' 'as.teal_slices' 'as.list.teal_slices'
     '[.teal_slices' 'c.teal_slices' 'format.teal_slices'
     'print.teal_slices'
   
   Functions with \usage entries need to have the appropriate \alias
   entries, and all their arguments documented.
   The \usage entries must correspond to syntactically valid R code.
   See chapter 'Writing R documentation files' in the 'Writing R
   Extensions' manual.
```

A dead end.

chlebowa commented:
Maybe there is a way to edit `~/R/x86_64-pc-linux-gnu-library/4.3/teal.slice/html/00Index.html` directly?

m7pr commented:
@chlebowa I tired with the INDEX file https://github.com/insightsengineering/teal.slice/issues/335#issuecomment-1973104763
The `index.html` probably gets created during package installation. Not a part of the source doe.

chlebowa commented:
Probably, yeah.
But `INDEX` is all right:
```
~/R/x86_64-pc-linux-gnu-library/4.3/teal.slice:$ more INDEX
FilterPanelAPI          Class to encapsulate the API of the filter
                        panel of a teal app
filter_state_api        Managing 'FilteredData' states
get_filter_expr         Gets filter expression for multiple 'datanames'
                        taking into account its order.
init_filtered_data      Initialize 'FilteredData'
teal_slice              Specify single filter
teal_slices             Complete filter specification
```

m7pr commented:
So I figured this INDEX  file does not have any influence on anything : P

m7pr commented:
I've got 2 more ideas. Will experiment with `@noRd` / `@rdname` tags, nor no tags for utility functions WITH removed `@aliases` tag.

m7pr commented:
My final idea was to (for each utility function) create a small documentation page

```r
#' @title `is.teal_slices`
#' @description Utility function for [`teal_slices`] object.
#' @export
#' @keywords internal
#'
```

that is not listed in package index (thanks to `@keywords internal`). 


If user hits `?teal_slices` (s)he will see an overview of `teal_slices` with a custom `@usage`. Is (s)he hits `?is.teal_slices` (s)he will have a change to get redirected directly to `?teal_slices`.

<img width="322" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/133694481/8d574f3b-81ec-4cc2-9bad-172ad7aa6228">

However **R CMD CHECK throws a warning about missing aliases for `teal_slices` **

```r
W  checking Rd \usage sections (854ms)
   Objects in \usage without \alias in documentation object 'teal_slices':
     'is.teal_slices' 'as.teal_slices' 'as.list.teal_slices'
     '[.teal_slices' 'c.teal_slices' 'format.teal_slices'
     'print.teal_slices'
```

m7pr commented:
I think I will need to take a break from this. Or we end up with a proposition from https://github.com/insightsengineering/teal.slice/issues/335#issuecomment-1976301898 where we have utility functions documented on a separate page, but they are linked from `?teal_slices` in the description part.
